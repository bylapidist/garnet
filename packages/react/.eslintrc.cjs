module.exports = {
  root: true,
  extends: ['@lapidist/garnet'],
};
