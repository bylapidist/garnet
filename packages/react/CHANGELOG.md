# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.5.48](https://github.com/bylapidist/garnet/compare/v0.5.47...v0.5.48) (2024-03-17)

**Note:** Version bump only for package @lapidist/garnet-react





## [0.5.45](https://github.com/bylapidist/garnet/compare/v0.5.44...v0.5.45) (2024-03-13)

**Note:** Version bump only for package @lapidist/garnet-react





## [0.5.41](https://github.com/bylapidist/garnet/compare/v0.5.40...v0.5.41) (2024-03-07)

**Note:** Version bump only for package @lapidist/garnet-react





## [0.5.36](https://github.com/bylapidist/garnet/compare/v0.5.35...v0.5.36) (2024-03-02)

**Note:** Version bump only for package @lapidist/garnet-react





## [0.5.33](https://github.com/bylapidist/garnet/compare/v0.5.32...v0.5.33) (2024-02-20)

**Note:** Version bump only for package @lapidist/garnet-react





## [0.5.29](https://github.com/bylapidist/garnet/compare/v0.5.28...v0.5.29) (2024-02-15)

**Note:** Version bump only for package @lapidist/garnet-react





## [0.5.22](https://github.com/bylapidist/garnet/compare/v0.5.21...v0.5.22) (2024-02-07)

**Note:** Version bump only for package @lapidist/garnet-react





## [0.5.17](https://github.com/bylapidist/garnet/compare/v0.5.16...v0.5.17) (2024-02-06)

**Note:** Version bump only for package @lapidist/garnet-react





## [0.5.4](https://github.com/bylapidist/garnet/compare/v0.5.3...v0.5.4) (2024-01-18)

**Note:** Version bump only for package @lapidist/garnet-react





## [0.5.2](https://github.com/bylapidist/garnet/compare/v0.5.1...v0.5.2) (2024-01-18)

**Note:** Version bump only for package @lapidist/garnet-react





# [0.5.0](https://github.com/bylapidist/garnet/compare/v0.4.23...v0.5.0) (2024-01-16)


### Features

* **apps/website:** add nextjs scaffolding to apps/website package ([ffca25b](https://github.com/bylapidist/garnet/commit/ffca25b89d56e470543b92cf3739fe3ce31c738e))





## [0.4.20](https://github.com/bylapidist/garnet/compare/v0.4.19...v0.4.20) (2024-01-12)

**Note:** Version bump only for package @lapidist/garnet-react





## [0.4.15](https://github.com/bylapidist/garnet/compare/v0.4.14...v0.4.15) (2024-01-05)

**Note:** Version bump only for package @lapidist/garnet-react





## [0.4.5](https://github.com/bylapidist/garnet/compare/v0.4.4...v0.4.5) (2023-12-17)

**Note:** Version bump only for package @lapidist/garnet-react





## [0.4.4](https://github.com/bylapidist/garnet/compare/v0.4.3...v0.4.4) (2023-12-16)

**Note:** Version bump only for package @lapidist/garnet-react





## [0.4.3](https://github.com/bylapidist/garnet/compare/v0.4.2...v0.4.3) (2023-12-16)

**Note:** Version bump only for package @lapidist/garnet-react





## [0.4.2](https://github.com/bylapidist/garnet/compare/v0.4.1...v0.4.2) (2023-12-16)


### Bug Fixes

* missing garnet-styles dependency ([3c2d6de](https://github.com/bylapidist/garnet/commit/3c2d6de947e6250aac3eb8b7b1382ec126196bbd))





# [0.4.0](https://github.com/bylapidist/garnet/compare/v0.3.4...v0.4.0) (2023-12-16)


### Features

* **styles:** implement styles and hot reloading ([3cc4cac](https://github.com/bylapidist/garnet/commit/3cc4cacb42ab61bbc43ee9b614c00705b113c46a))





## [0.3.4](https://github.com/bylapidist/garnet/compare/v0.3.3...v0.3.4) (2023-12-16)

**Note:** Version bump only for package @lapidist/garnet-react





# [0.3.0](https://github.com/bylapidist/garnet/compare/v0.2.23...v0.3.0) (2023-12-13)


### Features

* **tokens:** add tokens package ([53f13b3](https://github.com/bylapidist/garnet/commit/53f13b310df90fd37cd811a0a142efb3500bcb55))





## [0.2.3](https://github.com/bylapidist/garnet/compare/v0.2.2...v0.2.3) (2023-08-27)

**Note:** Version bump only for package @lapidist/garnet-react





# [0.2.0](https://github.com/bylapidist/garnet/compare/v0.1.2...v0.2.0) (2023-08-26)


### Features

* **storybook:** add storybook app ([b88dc90](https://github.com/bylapidist/garnet/commit/b88dc9008cd8ed325c96ee911e5209411b27994f))





## [0.1.2](https://github.com/bylapidist/garnet/compare/v0.1.1...v0.1.2) (2023-08-25)

**Note:** Version bump only for package @lapidist/garnet-react





## [0.1.1](https://github.com/bylapidist/garnet/compare/v0.1.1-beta.1...v0.1.1) (2023-08-24)

**Note:** Version bump only for package @lapidist/garnet-react





## [0.1.1-beta.1](https://github.com/bylapidist/garnet/compare/v0.1.1-beta.0...v0.1.1-beta.1) (2023-08-24)


### Bug Fixes

* **react:** add button type ([1c10ed5](https://github.com/bylapidist/garnet/commit/1c10ed5565034e6a4e1fb7bc3753707012eaafed))





## [0.1.1-beta.0](https://github.com/bylapidist/garnet/compare/v0.1.0...v0.1.1-beta.0) (2023-08-24)

**Note:** Version bump only for package @lapidist/garnet-react





# [0.1.0](https://github.com/bylapidist/garnet/compare/v0.1.0-beta.0...v0.1.0) (2023-08-24)

**Note:** Version bump only for package @lapidist/garnet-react





# 0.1.0-beta.0 (2023-08-24)

**Note:** Version bump only for package @lapidist/garnet-react
