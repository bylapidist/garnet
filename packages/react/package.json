{
  "name": "@lapidist/garnet-react",
  "version": "0.5.48",
  "description": "Lapidist components",
  "author": {
    "name": "Brett Dorrans",
    "email": "hello@lapidist.net"
  },
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/bylapidist/garnet.git"
  },
  "bugs": {
    "url": "https://github.com/bylapidist/garnet/issues"
  },
  "homepage": "https://github.com/bylapidist/garnet#readme",
  "publishConfig": {
    "access": "public"
  },
  "keywords": [
    "react",
    "garnet",
    "lapidist",
    "components",
    "component library",
    "design system"
  ],
  "type": "module",
  "files": [
    "dist",
    "CHANGELOG.md"
  ],
  "main": "./dist/index.umd.js",
  "module": "./dist/index.es.js",
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "require": "./dist/index.umd.js",
      "import": "./dist/index.es.js"
    },
    "./style.css": {
      "import": "./dist/style.css",
      "require": "./dist/style.css"
    }
  },
  "scripts": {
    "build": "tsc && vite build",
    "watch": "concurrently npm:watch:ts npm:watch:vite",
    "watch:ts": "tsc -w",
    "watch:vite": "vite build --watch",
    "watch:tests": "vitest",
    "lint": "eslint .",
    "fix": "eslint . --fix",
    "test": "vitest run --coverage",
    "typecheck": "tsc --noEmit -p tsconfig.json",
    "snapshots": "vitest run --update"
  },
  "dependencies": {
    "@lapidist/garnet-styles": "^0.5.48",
    "@lapidist/garnet-tokens": "^0.5.4",
    "react": "18.x",
    "react-dom": "18.x"
  },
  "devDependencies": {
    "@lapidist/garnet-types": "^0.5.0"
  },
  "peerDependencies": {
    "react": "18.x",
    "react-dom": "18.x"
  }
}
