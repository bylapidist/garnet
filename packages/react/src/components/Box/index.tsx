import React from 'react';
import cx from 'classnames';
import { CommonProps, BoxProps } from '@lapidist/garnet-types';

import styles from '@lapidist/garnet-styles/components/box.module.css';

export default function Box({
    id,
    className,
    as: Component = 'div',
    testId = 'Box',
    elevation = 'none',
    gutter = 'none',
    roundness = 'none',
    ...restProps
}: React.PropsWithChildren<CommonProps<BoxProps>>) {
    return (
        <Component
            data-testid={testId}
            id={id}
            className={cx(
                styles[`elevation-${elevation}`],
                styles[`roundness-${roundness}`],
                styles[`gutter-${gutter}`],
                className,
            )}
            {...restProps}
        />
    );
}
