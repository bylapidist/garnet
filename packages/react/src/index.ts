export { default as App } from './components/App';
export { default as Button } from './components/Button';
export { default as Text } from './components/Text';
