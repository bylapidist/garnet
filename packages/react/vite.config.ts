/// <reference types="vitest" />
import react from '@vitejs/plugin-react';
import path from 'path';
import { defineConfig } from 'vite';
import dts from 'vite-plugin-dts';

// https://vitejs.dev/config/
export default defineConfig({
    plugins: [
        react(),
        dts({
            insertTypesEntry: true,
        }),
    ],
    test: {
        globals: true,
        environment: 'jsdom',
        coverage: {
            provider: 'v8',
        },
    },
    build: {
        lib: {
            entry: path.resolve(__dirname, 'src/index.ts'),
            name: 'garnet-react',
            formats: ['es', 'umd'],
            fileName: (format) => `index.${format}.js`,
        },
        rollupOptions: {
            external: ['react', 'react-dom'],
            output: {
                globals: {
                    react: 'React',
                    'react-dom': 'ReactDOM',
                },
            },
        },
    },
});
