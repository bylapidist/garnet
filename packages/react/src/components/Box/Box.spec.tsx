import React from 'react';
import { describe, expect, test, afterEach } from 'vitest';
import { cleanup, render, screen } from '@testing-library/react';
import Box from './index';
import { NullableSize } from '@lapidist/garnet-types/common';

afterEach(cleanup);

const setup = (Component: React.ReactElement) => render(Component);

const sizes: NullableSize[] = ['none', 'small', 'medium', 'large'];

describe('Box', () => {
    test('it matches snapshot', () => {
        const { container } = setup(<Box>Hello world</Box>);
        expect(container.firstChild).toMatchSnapshot();
    });

    test.each(sizes)('it works with gutters', (gutter) => {
        const { container } = setup(<Box gutter={gutter}>Hello world</Box>);
        expect(container.firstChild).toMatchSnapshot();
    });

    test.each(sizes)('it works with elevations', (elevation) => {
        const { container } = setup(
            <Box elevation={elevation}>Hello world</Box>,
        );
        expect(container.firstChild).toMatchSnapshot();
    });

    test.each(sizes)('it works with roundness', (roundness) => {
        const { container } = setup(
            <Box roundness={roundness}>Hello world</Box>,
        );
        expect(container.firstChild).toMatchSnapshot();
    });

    test('it works with default testId', () => {
        setup(<Box gutter={sizes[0]}>Hello world</Box>);
        expect(screen.getByTestId('Box')).toBeTruthy();
    });

    test('it works with specified testId', () => {
        setup(
            <Box gutter={sizes[0]} testId={'TestId'}>
                Hello world
            </Box>,
        );
        expect(screen.getByTestId('TestId')).toBeTruthy();
    });

    test('it works with as', () => {
        setup(
            <Box gutter={sizes[0]} as="span">
                Hello world
            </Box>,
        );
        expect(screen.getByTestId('Box').nodeName).toBe('SPAN');
    });

    test('it works with className', () => {
        setup(
            <Box gutter={sizes[0]} className="test">
                Hello world
            </Box>,
        );
        expect(
            screen.getByTestId('Box').classList.contains('test'),
        ).toBeTruthy();
    });
});
