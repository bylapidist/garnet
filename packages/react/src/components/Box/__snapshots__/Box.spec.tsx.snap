// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Box > it matches snapshot 1`] = `
<div
  class="_elevation-none_02cc31 _roundness-none_02cc31 _gutter-none_02cc31"
  data-testid="Box"
>
  Hello world
</div>
`;

exports[`Box > it works with elevations 1`] = `
<div
  class="_elevation-none_02cc31 _roundness-none_02cc31 _gutter-none_02cc31"
  data-testid="Box"
>
  Hello world
</div>
`;

exports[`Box > it works with elevations 2`] = `
<div
  class="_elevation-small_02cc31 _roundness-none_02cc31 _gutter-none_02cc31"
  data-testid="Box"
>
  Hello world
</div>
`;

exports[`Box > it works with elevations 3`] = `
<div
  class="_elevation-medium_02cc31 _roundness-none_02cc31 _gutter-none_02cc31"
  data-testid="Box"
>
  Hello world
</div>
`;

exports[`Box > it works with elevations 4`] = `
<div
  class="_elevation-large_02cc31 _roundness-none_02cc31 _gutter-none_02cc31"
  data-testid="Box"
>
  Hello world
</div>
`;

exports[`Box > it works with gutters 1`] = `
<div
  class="_elevation-none_02cc31 _roundness-none_02cc31 _gutter-none_02cc31"
  data-testid="Box"
>
  Hello world
</div>
`;

exports[`Box > it works with gutters 2`] = `
<div
  class="_elevation-none_02cc31 _roundness-none_02cc31 _gutter-small_02cc31"
  data-testid="Box"
>
  Hello world
</div>
`;

exports[`Box > it works with gutters 3`] = `
<div
  class="_elevation-none_02cc31 _roundness-none_02cc31 _gutter-medium_02cc31"
  data-testid="Box"
>
  Hello world
</div>
`;

exports[`Box > it works with gutters 4`] = `
<div
  class="_elevation-none_02cc31 _roundness-none_02cc31 _gutter-large_02cc31"
  data-testid="Box"
>
  Hello world
</div>
`;

exports[`Box > it works with roundness 1`] = `
<div
  class="_elevation-none_02cc31 _roundness-none_02cc31 _gutter-none_02cc31"
  data-testid="Box"
>
  Hello world
</div>
`;

exports[`Box > it works with roundness 2`] = `
<div
  class="_elevation-none_02cc31 _roundness-small_02cc31 _gutter-none_02cc31"
  data-testid="Box"
>
  Hello world
</div>
`;

exports[`Box > it works with roundness 3`] = `
<div
  class="_elevation-none_02cc31 _roundness-medium_02cc31 _gutter-none_02cc31"
  data-testid="Box"
>
  Hello world
</div>
`;

exports[`Box > it works with roundness 4`] = `
<div
  class="_elevation-none_02cc31 _roundness-large_02cc31 _gutter-none_02cc31"
  data-testid="Box"
>
  Hello world
</div>
`;
